pulsar-common/src/main/java/org/apache/pulsar/common/util/collections/ConcurrentLongHashMap.java

```diff
@@ -237,6 +238,12 @@ public void clear() {
         }
     }
 
+    /**
+     * Iterate over all the entries in the map and apply the processor function to each of them.
+     * <p>
+     * <b>Warning: Do Not Guarantee Thread-Safety.</b>
+     * @param processor the processor to apply to each entry
+     */
```

Comment:
Rather than simply stating "Warning: Do Not Guarantee Thread-Safety," it would be more informative to provide a detailed description of the actual behavior, including what is guaranteed and what is not in terms of thread-safety.

Reply:
What about:
```
Method forEach are designed to be used by only one thread at a time. When iterate the map with concurrent write, it is possible to see or not to see the new value. There is no gurantee that if we write value1,value2 and we can see the value2, then we can see value1.  We may even encounter two mappings with same key, thus method keys may return a List containg two identical keys.
Bear in mind that the results of aggregate status methods including keys, values, asMap are typically useful only when a map is not undergoing concurrent updates in other threads. Otherwise the results of these methods reflect transient states that may be adequate for monitoring or estimation purposes, but not for program control. 
```

Reply:
updated. @codelipenghui